src/CMakeLists.txt

```diff
@@ -42,18 +65,34 @@ function(pico_add_map_output TARGET)
     endif ()
 endfunction()
 
-# create a hardware_NAME_headers target (see pico_pico_simple_hardware_headers_target)
-# create a hardware_NAME target (see pico_pico_simple_hardware_target)
+# create a hardware_NAME_headers target (see pico_simple_hardware_headers_target)
+# create a hardware_NAME target (see pico_simple_hardware_target)
 macro(pico_simple_hardware_target NAME)
     pico_simple_hardware_headers_target(${NAME})
     pico_simple_hardware_impl_target(${NAME})
 endmacro()
 
 # create an INTERFACE library named target, and define LIB_TARGET=1 (upper case) as a compile option
+# and make it dependent on a pre-existing corresponding _headers library
+# optional arg NOFLAG will skip the LIB_TARGET definition
 function(pico_add_impl_library target)
     add_library(${target} INTERFACE)
     string(TOUPPER ${target} TARGET_UPPER)
-    target_compile_definitions(${target} INTERFACE LIB_${TARGET_UPPER}=1)
+    if (${ARGC} GREATER 1)
+        if (NOT "${ARGV1}" STREQUAL "NOFLAG")
+            message(FATAL_ERROR "Unknown parameter ${ARGV1}")
+        endif()
+    else()
+        target_compile_definitions(${target} INTERFACE LIB_${TARGET_UPPER}=1)
```

Comment:
You might have to explain to me what these LIB_* defs are for

Reply:
The idea was to allow inter-dependent libraries to be aware of each other, and to change/add behavior or guard includes

so you can do 

```
#if LIB_PICO_FOO
#include "pico/foo.h"
#endif
```

as well - you can search on them; they are using a lot in the `stdio` stuff and with some of the swapable libraries. I'm kind of leaning in favor of making these more explicit than implicit, and thus have also started adding the new `NOFLAG` option to avoid setting them... of course it is hard to know who has taken advantage of them, but probably few if any beyond SDK and examples/extras/playground

Note: you might think (as did I) the the LIB_ flag therefore belongs on the header, however in practice we only check them in the C files